plugin.py

```diff
@@ -0,0 +1,154 @@
+    control_ip = c.config["control_node"]
+
+    print "Generating plugin certs"
+    # generate and upload plugin.crt and plugin.key for each node
+    for node in c.config["agent_nodes"]:
+        
+        # use the node IP to name the local files so they do not overwrite each other
+        c.run("flocker-ca create-api-certificate %s" % (node + '-plugin',))
+        print "Generated plugin certs for", node
+        # upload the .crt and .key
+        for ext in ("crt", "key"):
+            c.scp("%s-plugin.%s" % (node, ext),
+                node, "/etc/flocker/plugin.%s" % (ext,))
+        print "Uploaded plugin certs for", node
+
+    print "Installing flocker plugin"
+    # loop each agent and get the plugin installed/running
+    # clone the plugin and configure an upstart/systemd unit for it to run
+    for node in c.config["agent_nodes"]:
+
+        # we need this so we know what folder to cd into
+        plugin_repo_folder = settings['PLUGIN_REPO'].split('/').pop()
+
+        # the full api path to the control service
+        controlservice = 'https://%s:4523/v1' % (control_ip)
+        c.runSSHRaw(node, "rm -rf %s" % (plugin_repo_folder))
+        # clone the right repo and checkout the branch
+        print "Cloning the plugin repo on %s - %s" % (node, settings['PLUGIN_REPO'])
+        c.runSSHRaw(node, "git clone -b %s %s || true" % (settings['PLUGIN_BRANCH'], settings['PLUGIN_REPO']))
+
+        # install pip and python-dev
+        if c.config["os"] == "ubuntu":
+            c.runSSHRaw(node, "apt-get install -y python-dev python-pip")
+        # configure a systemd job that runs the bash script
+        elif c.config["os"] == "centos":
+            c.runSSHRaw(node, "yum install -y python-devel python-pip")
+
+        # pip install the plugin
+        c.a(node, "pip install -r /root/%s/requirements.txt" % (plugin_repo_folder))
+        
+        print "Have control service: %s" % (controlservice)
+        # a bash script that runs the app via twistd
+        # this makes the upstart - systemd files much easier shorter
+        print "Writing runflockerplugin.sh to %s" % (node)
+        c.runSSH(node, """cat << EOF > /root/runflockerplugin.sh
+#!/usr/bin/env bash
+rm -f /usr/share/docker/plugins/flocker.sock || true
+export FLOCKER_CONTROL_SERVICE_BASE_URL=%s
+export MY_NETWORK_IDENTITY=%s
+export USER_CERTIFICATE_FILENAME=plugin.crt
+export USER_KEY_FILENAME=plugin.key
+cd /root/%s && twistd -noy powerstripflocker.tac
+EOF
+""" % (controlservice, node, plugin_repo_folder))
+
+        # configure an upstart job that runs the bash script
+        if c.config["os"] == "ubuntu":
+            print "Writing flocker-plugin upstart job to %s" % (node)
+            c.runSSH(node, """cat <<EOF > /etc/init/flocker-plugin.conf
+# flocker-plugin - flocker-plugin job file
+
+description "Flocker Plugin service"
+author "ClusterHQ <support@clusterhq.com>"
+
+respawn
+
+# Start the process
+exec /usr/bin/env bash /root/runflockerplugin.sh
```

Comment:
I'd rather that this just ran the flocker plugin. Can you move the env var configuration into the upstart script?